src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -94,39 +102,18 @@ private enum Flags : byte
             _containingType = containingType;
             _refKind = refKind;
 
-            SyntaxTokenList modifiers = GetModifierTokens(syntax);
             if (binder is object)
             {
-                binder = binder.WithUnsafeRegionIfNecessary(modifiers);
+                binder = binder.WithUnsafeRegionIfNecessary(GetModifierTokens(syntax));
                 binder = binder.WithAdditionalFlagsAndContainingMemberOrLambda(BinderFlags.SuppressConstraintChecks, this);
             }
 
-            var arrowExpression = GetArrowExpression(syntax);
-            bool hasExpressionBody = arrowExpression != null;
-            bool hasInitializer = HasInitializer(syntax);
-
-            GetAccessorDeclarations(
-                syntax,
-                diagnostics,
-                out bool isAutoProperty,
-                out bool hasAccessorList,
-                out bool accessorsHaveImplementation,
-                out bool isInitOnly,
-                out var getSyntax,
-                out var setSyntax);
-
-            var hasGetAccessor = getSyntax is object;
-            var hasSetAccessor = setSyntax is object;
-
-            bool modifierErrors;
-            _modifiers = MakeModifiers(modifiers, isExplicitInterfaceImplementation, isIndexer,
-                                       accessorsHaveImplementation, location,
-                                       diagnostics, out modifierErrors);
```

Comment:
I am not comfortable with removal of this code. Anything that deals with diagnostics should stay here for now. derived types shouldn't be in the business of deciding what checks should be performed

Reply:
> I am not comfortable with removal of this code. 

On second thought, that is probably a change worth making. 

---
In reply to: [446524304](https://github.com/dotnet/roslyn/pull/45336#discussion_r446524304) [](ancestors = 446524304)